src/main/java/Duke.java

```diff
@@ -1,10 +1,213 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+
 public class Duke {
+    //public static Task[] tasks = new Task[100];
+    public static ArrayList<Task> tasks = new ArrayList<>();
+
     public static void main(String[] args) {
+
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
+
+        // print welcome message
         System.out.println("Hello from\n" + logo);
+        System.out.println("-".repeat(50));
+        System.out.println("\tHello! I'm Duke");
+        System.out.println("\tWhat can I do for you?");
+        printDash();
+        String pathOfFile = new File("").getAbsolutePath();
+        File tasksFile = new File(pathOfFile + "/duke.txt");
+        try {
+            if (tasksFile.createNewFile()) {
+                System.out.println("\tTo save your task locally,\n" +
+                        "\tA new file has been created at:\n\t" +
+                        tasksFile.getAbsolutePath() + "\n");
+            }
+            uploadTasks(tasksFile.getAbsolutePath());
+        } catch (IOException e) {
+            System.out.println("\tThere was an I/O error:\nBye!\n");
+            e.printStackTrace();
+        }
+        Scanner myObj = new Scanner(System.in);
+        String command = myObj.nextLine();
+
+        while (!command.equals("bye")) {
+            // prints a checklist
+            if (command.equals("list")) {
+                printList();
+
+                // marks current task as completed
+            } else if (command.contains("done")) {
+                taskCompleted(command);
+
+
+                // adds tasks into list
+            } else if (command.contains("delete")) {
+                deleteTasks(command);
+
+                // add tasks to list
+
+            } else {
+                addTasks(command);
+            }
+            command = myObj.nextLine();
+        }
+        // exit program
+        printDash();
+        System.out.println("Bye. Hope to see you again soon!");
+        printDash();
+        try {
+            saveTasks(tasksFile.getAbsolutePath());
+        } catch (IOException e) {
+            System.out.println("No file was saved due to an I/O error.\n");
+        }
+
     }
+    public static void uploadTasks(String pathOfFile) throws FileNotFoundException {
+        File f = new File(pathOfFile);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String[] task = s.nextLine().split(" ==> ");
+            if (task[0].equals("T")) {
```

Comment:
to improve the readability, maybe assign task[0] to some variable? e.g String command = task[0]; Same goes for task[2], tasks[3]...Quite difficulty to understand the meaning.